src/planners/path_search.jl

```diff
@@ -56,14 +56,53 @@ function Base.copy(sol::PathSearchSolution)
                               search_tree, search_frontier, search_order)
 end
 
-Base.iterate(sol::PathSearchSolution) = iterate(sol.plan)
-Base.iterate(sol::PathSearchSolution, istate) = iterate(sol.plan, istate)
-Base.getindex(sol::PathSearchSolution, i::Int) = getindex(sol.plan, i)
-Base.length(sol::PathSearchSolution) = length(sol.plan)
+"""
+Solution type for search-based planners that produce fully ordered plans.
+"""
+mutable struct BiPathSearchSolution{S<:State,T} <: OrderedSolution
+    status::Symbol
+    plan::Vector{Term}
+    trajectory::Union{Vector{S},Nothing}
+    expanded::Int
+    f_search_tree::Union{Dict{UInt,PathNode{S}},Nothing}
+    f_frontier::T
+    f_expanded::Int
+    f_trajectory::Union{Vector{S},Nothing}
+    b_search_tree::Union{Dict{UInt,PathNode{S}},Nothing}
+    b_frontier::T
+    b_expanded::Int
+    b_trajectory::Union{Vector{S},Nothing}
+end
+
+BiPathSearchSolution(status::Symbol, plan) =
+    BiPathSearchSolution(status, plan, nothing, -1, nothing, nothing, -1, nothing, nothing, nothing, -1, nothing)
+BiPathSearchSolution(status::Symbol, plan, trajectory) =
+    BiPathSearchSolution(status, plan, trajectory, -1, nothing, nothing, -1, nothing, nothing, nothing, -1, nothing)
+
+function Base.copy(sol::BiPathSearchSolution)
+    fields = map(fieldnames(BiPathSearchSolution)) do field 
+        x = getfield(sol, field)
+        (x isa Symbol || isnothing(x)) && return(x)
+        copy(x)
+    end
+    return BiPathSearchSolution(fields...)
+end
+
+const SearchSolutions = Union{PathSearchSolution, BiPathSearchSolution}
```

Comment:
Let's replace with this with an abstract type called `AbstractPathSearchSolution` (I don't want to call it `AbstractSearchSolution` because there are other kinds of search algorithms that don't necessarily fit this mold.)

Reply:
Oh, and let's make sure that `AbstractPathSearchSolution` is a subtype of `OrderedSolution`.

Reply:
Done